### ISSUE TITLE
xmlns:svgjs namespace added even if it exists already

### ISSUE THREAD

**Issue by jlherren** [svgdotjs/svg.js#709]
SVG.js adds the 'svgjs' namespace to the `<svg>` element, but it does so even when it is there already. When importing a previously exported SVG this results in a duplicate attribute. Chrome will refuse to display such an SVG file when opened stand-alone, because it is not valid XML. (It seems to work fine when embedded in HTML5, since this won't trigger strict XML mode). Here's an example:

```
<!DOCTYPE html>
<html>
<head>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/svg.js/2.6.2/svg.min.js"></script>
    <script>
        window.onload = function () {
            // new drawing
            var draw = SVG('container').size(100, 100);
            draw.line(90, 10, 10, 90).stroke({color:'#ff8000', width:20});

            // export into str
            var str = draw.svg();
            console.log('Got this SVG', str);

            // reimport into a second drawing
            document.getElementById('container2').insertAdjacentHTML('beforeend', str);
            var draw2 = SVG(document.getElementById('container2').firstChild).size(100, 100);

            // export second drawing
            var str2 = draw2.svg();
            console.log('Got this SVG', str2);
        };
    </script>
</head>
<body>

<div id="container"></div>
<div id="container2"></div>

</body>
</html>
```
As is visible in the dev console, the second export has two `xmlns:svgjs="..."` attributes on the `<svg>`.

(Tested on Chrome only.)

**Comment by Fuzzyma** [comment]
Please put this in a fiddle. That would help a lot!

**Comment by jlherren** [comment]
Here you go: https://jsfiddle.net/29wnw3da/1/

**Comment by Fuzzyma** [comment]
Thats odd... its all created the same way: 

```js
    namespace: function() {
      return this
        .attr({ xmlns: SVG.ns, version: '1.1' })
        .attr('xmlns:xlink', SVG.xlink, SVG.xmlns)
        .attr('xmlns:svgjs', SVG.svgjs, SVG.xmlns)
    }
```

**Comment by jlherren** [comment]
I tried to trace through the code and it seems that the call to `this.node.setAttributeNS()` (inside `attr()`) indeed inserts a duplicate, which might hint at a browser bug. But I can't reproduce that behavior with a smaller example.

It's worth noting that Firefox behaves the same way as Chrome, whereas Internet Explorer does not have this problem. IE doesn't even output any of those additional namespaces at all (only the default namespace), which is actually a quite desirable feature, since my SVGs don't need the 'xlink' and 'svgjs' namespaces in the first place (I'm not even sure why they are inserted).

I will just work around the problem by manually removing the `xmlns:svgjs` attribute before importing.

**Comment by Fuzzyma** [comment]
To use any image, use or other linked element in svg you need the xlink namespace. This is removed im svg2 (which is not really a good thing). For the svgjs namespace: well we save data on the Dom too. It wouldn't be valid XML without it

**Comment by jlherren** [comment]
In Internet Explorer 11 it's even worse: Ehe `xmlns` attribute is duplicate on a trivially empty drawing. See https://jsfiddle.net/48r4mf98/1/

**Comment by jlherren** [comment]
Seems to be an IE11 problem. [Highcharts was having the same problem](https://github.com/highcharts/highcharts/issues/1978) and apparently implemented a work around.

**Comment by Fuzzyma** [comment]
@jlherren can you please checkout the 3.0 branch and check if the problem persists?
Make sure to change your code according to the breaking changes e.g.

```js
draw = SVG(node) 
//becomes
draw = SVG().addTo(node)
```

**Comment by jlherren** [comment]
I inserted the current 3.0.0 branch into my original code from the first post with the necessary adaptations and it seems to work better now: No more duplicate namespaces, yay!

However, the second exported SVG is missing the `<line>` element now. But I suppose that's a separate and unrelated issue. For reference here's the output I got with 3.0.0:

```
Got this SVG <svg xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs" width="100" height="100"><defs></defs><line x1="90" y1="10" x2="10" y2="90" stroke-width="20" stroke="#ff8000"></line></svg>
Got this SVG <svg xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs" width="100" height="100"><defs></defs></svg>
```

(I'd create a plunker but I don't know how to include a dev version of a library)

**Comment by Fuzzyma** [comment]
Can you show me the code you used to create that svg? Could he that api changes lead to this behavior

**Comment by jlherren** [comment]
That's what I did. `svg.min.js` is the current HEAD of the `3.0.0` branch. As you can see in the dev console, the second export is wrong even though the import was successful (the second drawing appears correctly).

```
<!DOCTYPE html>
<html>
<head>
    <script src="svg.min.js"></script>
    <script>
        window.onload = function () {
            // new drawing
            var draw = SVG().addTo(document.getElementById('container')).size(100, 100);
            draw.line(90, 10, 10, 90).stroke({color:'#ff8000', width:20});

            // export into str
            var str = draw.svg();
            console.log('Got this SVG', str);

            // reimport into a second drawing
            document.getElementById('container2').insertAdjacentHTML('beforeend', str);
            var draw2 = SVG().addTo(document.getElementById('container2').firstChild).size(100, 100);

            // export second drawing
            var str2 = draw2.svg();
            console.log('Got this SVG', str2);
        };
    </script>
</head>
<body>

<div id="container"></div>
<div id="container2"></div>

</body>
</html>
```

**Comment by Fuzzyma** [comment]
So you say the line is correctly imported but does not show up in the exported svg? That's odd. Need to investigate this

**Comment by Fuzzyma** [comment]
Ok - I just found that you just used that api a bit wrong. You insert the new svg manually (with native code) and then you create a blank new svg and append this to the container, too. No wonder this new svg is empty :).
The right syntax would be:

```js
draw2 = SVG(str).addTo('#container')
```

Looks nice, doesnt it? :)

**Comment by lalop** [comment]
@Fuzzyma I have the same issue and tried to fix the jsfiddle exemple given by @jlherren but it's still broken as you can see https://jsfiddle.net/y4n1se8t/

I replaced 
```
document.getElementById('container2').insertAdjacentHTML('beforeend', str);
var draw2 = SVG(document.getElementById('container2').firstChild);
```
by
```
var draw2 = SVG(str).addTo('#container2');
```

Am I wrong ?

**Comment by Fuzzyma** [comment]
Its stated in this issue, that the syntax you try to use is 3.0 syntax but you included 2.6 in the fiddle.
Read carefully trough the issue before posting!

If the issue remains, open a new bug report whith a minimal example fiddle showing the bug

**Comment by lalop** [comment]
sorry i didn't pay attention at the library version in jsfiddke
